JSF throws ClassNotFoundException for o.a.m.el.convert.ValueExpressionToValueBinding

**Describe the bug**  
A JSF application using `jsf-2.2` or `jsf-2.3` can encounter a `ClassNotFoundException` for the internal MyFaces class `ValueExpressionToValueBinding`, even if that class isn't directly referenced by the app. This issue occurs because the `org.apache.myfaces.el.convert` package is not currently made available (by Liberty's `jsf-2.2` and `jsf-2.3` bundles) to the thread context classloader. 

```
[ERROR   ] SRVE0315E: An exception occurred: java.lang.Throwable: javax.servlet.ServletException: org.apache.myfaces.el.convert.ValueExpressionToValueBinding
        at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5119)
        at [internal classes]
Caused by: javax.servlet.ServletException: org.apache.myfaces.el.convert.ValueExpressionToValueBinding
        at javax.faces.webapp.FacesServlet.service(FacesServlet.java:230)
        ... 1 more
Caused by: java.lang.ClassNotFoundException: org.apache.myfaces.el.convert.ValueExpressionToValueBinding
        at java.lang.Class.forNameImpl(Native Method)
        at java.lang.Class.forName(Class.java:407)
        at javax.faces.component._ClassUtils.classForName(_ClassUtils.java:145)
        ... 1 more
```

**Steps to Reproduce**  
Applications with custom tags using [Application.createValueBinding(String)](https://docs.oracle.com/javaee/6/api/javax/faces/application/Application.html#createValueBinding(java.lang.String)) deployed with `jsf-2.2` or `jsf-2.3`can hit this:

```
@FacesComponent(tagName = "valueBindingTag", createTag=true, namespace="http://openliberty.io/jsf")
public class ValueBindingTag extends UIComponentBase {
...
    @Override
    public void encodeBegin(FacesContext context) throws IOException {
        ValueBinding test = getFacesContext().getApplication().createValueBinding("test"); // CNFE here
        this.setValueBinding("test", test);
    }
...
```

**Expected behavior**  
This `ClassNotFoundException` should not be thrown

**Diagnostic information:**  
 - OpenLiberty Version: [e.g. 21.0.0.8 - 21.0.0.10]
 - Affected feature(s) `jsf-2.2`, `jsf-2.3`

**Additional context**  
N/A

--- a/dev/com.ibm.ws.jsf.2.2/bnd.bnd
+++ b/dev/com.ibm.ws.jsf.2.2/bnd.bnd
@@ -91,6 +91,7 @@ Export-Package: \
   org.apache.myfaces.context.servlet;thread-context=true, \
   org.apache.myfaces.config.annotation;thread-context=true, \
   org.apache.myfaces.config.impl.digester.elements;thread-context=true, \
+  org.apache.myfaces.el.convert;thread-context=true,\
   org.apache.myfaces.flow;thread-context=true, \
   org.apache.myfaces.flow.cdi;thread-context=true, \
   org.apache.myfaces.lifecycle;thread-context=true, \

--- a/dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22MiscellaneousTests.java
+++ b/dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22MiscellaneousTests.java
@@ -11,6 +11,7 @@
 package com.ibm.ws.jsf22.fat.tests;
 
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
 
 import java.net.URL;
 
@@ -31,6 +32,7 @@ import com.gargoylesoftware.htmlunit.html.HtmlElement;
 import com.gargoylesoftware.htmlunit.html.HtmlPage;
 import com.gargoylesoftware.htmlunit.html.HtmlTextInput;
 import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
 import com.ibm.ws.jsf22.fat.JSFUtils;
 
 import componenttest.annotation.Server;
@@ -485,6 +487,32 @@ public class JSF22MiscellaneousTests {
                 Assert.fail("index.xhtml did not render properly.");
             } else if (!page.asText().contains("Test Success")) {
                 Assert.fail("/FacesConfigMissingXmlns/index.xhtml did not contain \"Test Success\":\n" + page.asXml());
+            }
+        }
+    }
+
+    /**
+     *
+     * Check that a ClassNotFoundException is not thrown when a custom tag uses Application.createValueBinding() 
+     * See https://github.com/OpenLiberty/open-liberty/issues/18437
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testCustomValueBindingTag() throws Exception {
+        try (WebClient webClient = new WebClient()) {
+            webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
+
+            URL url = JSFUtils.createHttpUrl(jsf22MiscellaneousServer, contextRootMiscellaneous, "testCustomValueBindingTag.jsf");
+            HtmlPage page = (HtmlPage) webClient.getPage(url);
+            page = page.getElementById("form:submit").click();
+
+            if (page == null) {
+                Assert.fail("testCustomValueBindingTag.xhtml did not render properly.");
+            } else {
+                Log.info(c, "page text:", page.asText());
+                assertFalse("A ClassNotFoundException was thrown", page.asText().contains("java.lang.ClassNotFoundException"));
+                assertTrue("Unexpected output", page.asText().contains("18437"));
             }
         }
     }

--- a/dev/com.ibm.ws.jsf.2.2_fat/test-applications/JSF22Miscellaneous.war/resources/testCustomValueBindingTag.xhtml
+++ b/dev/com.ibm.ws.jsf.2.2_fat/test-applications/JSF22Miscellaneous.war/resources/testCustomValueBindingTag.xhtml
@@ -1,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ -->
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:h="http://xmlns.jcp.org/jsf/html"
+      xmlns:test="http://openliberty.io/jsf">
+
+    <h:body>
+        <h:form id="form">
+            <test:valueBindingTag/>
+            Submitting this page will cause a ClassNotFoundException without the fix in 
+            https://github.com/OpenLiberty/open-liberty/issues/18437
+            <h:commandButton id="submit" value="submit"/>
+        </h:form>
+    </h:body>
+</html>

--- a/dev/com.ibm.ws.jsf.2.2_fat/test-applications/JSF22Miscellaneous.war/src/com/ibm/ws/jsf22/fat/miscbean/ValueBindingTag.java
+++ b/dev/com.ibm.ws.jsf.2.2_fat/test-applications/JSF22Miscellaneous.war/src/com/ibm/ws/jsf22/fat/miscbean/ValueBindingTag.java
@@ -1,0 +1,37 @@
+/*
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ */
+package com.ibm.ws.jsf22.fat.miscbean;
+
+import java.io.IOException;
+import java.util.Hashtable;
+
+import javax.faces.component.FacesComponent;
+import javax.faces.component.UIComponentBase;
+import javax.faces.context.FacesContext;
+import javax.faces.el.ValueBinding;
+
+@FacesComponent(tagName = "valueBindingTag", createTag=true, namespace="http://openliberty.io/jsf")
+public class ValueBindingTag extends UIComponentBase {
+
+    public ValueBindingTag() {
+    }
+
+    @Override
+    public void encodeBegin(FacesContext context) throws IOException {
+        ValueBinding test = getFacesContext().getApplication().createValueBinding("test");
+        this.setValueBinding("test", test);
+    }
+
+    @Override
+    public String getFamily() {
+        return "test";
+    }
+}

--- a/dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd
+++ b/dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd
@@ -87,6 +87,7 @@ Export-Package: \
   org.apache.myfaces.context.servlet;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.config.annotation;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.config.impl.digester.elements;thread-context=true;version=${Implementation-Version}, \
+  org.apache.myfaces.el.convert;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.flow;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.flow.cdi;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.lifecycle;thread-context=true;version=${Implementation-Version}, \

--- a/dev/io.openliberty.org.apache.myfaces.3.0/bnd.bnd
+++ b/dev/io.openliberty.org.apache.myfaces.3.0/bnd.bnd
@@ -77,6 +77,7 @@ Export-Package: \
   org.apache.myfaces.context.servlet;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.config.annotation;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.config.impl.digester.elements;thread-context=true;version=${Implementation-Version}, \
+  org.apache.myfaces.el.convert;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.flow;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.flow.cdi;thread-context=true;version=${Implementation-Version}, \
   org.apache.myfaces.lifecycle;thread-context=true;version=${Implementation-Version}, \

